### ISSUE TITLE
GSSAPI-Support for OpenLDAP

### ISSUE THREAD

**Issue by Oleg-Mishenkin** [einfallstoll/express-ntlm#13]
When connecting to OpenLDAP server there is an error in response that GSS-SPNEGO not supported on OpenLDAP server side. Can I change authentication mode to GSSAPI for example?

**Comment by einfallstoll** [comment]
SPNEGO ("Simple and Protected GSSAPI Negotiation Mechanism") is a mechanism to negotiate the choice of security method. It's useful if two ends want to communicate to each other but don't know what mechanism the other end "speaks".

From [How LDAP Authentication works with Active Directory and the UWWI](http://www.netid.washington.edu/documentation/ldapauth.aspx):
> The GSS-SPNEGO authentication mechanism (RFC4178) is actually the GSSAPI authentication mechanism but with a client-server negotiation mechanism that provides for selection of the preferred security mechanism that both client and server support. In this case, the server will prefer Kerberos then NTLMv2 then NTLM (UWWI does not support LM). For that reason, refer to the GSSAPI authentication mechanism for further details.

**Comment by Oleg-Mishenkin** [comment]
What will happen when LDAP server does not support neither GSS-SPNEGO nor GSSAPI by default as in my case? An negotiation error will occur. I see ASN1.makeoctstr('GSS-SPNEGO') in your source code at NTLM_AD_Proxy.js . May it be better for ability to customize authentication method? Thank you!

**Comment by einfallstoll** [comment]
I agree that `GSSAPI` makes sense to support. Although it will probably need some serious changes to the code. *Keep in mind, that I officially only support Active Directory so far.*

I marked this as an enhancement, so I can add `GSSAPI`-support in a future release.

**Comment by einfallstoll** [comment]
Are there any other mechanisms apart from `GSS-SPNEGO` and `GSSAPI` to support?
